### PR TITLE
Add KUBECON.md

### PR DESCRIPTION
**Pull request by kingdonb**
This is associated with fluxcd/website#1804

It will not be published on the website before we merge fluxcd/website#1804

We can see that it works in the preview build, (but we will need to merge this one first before we can see it working in the preview.)

I don't see why we would publish this page any earlier than ~1-2 weeks before KubeCon at earliest.

But there may be no reason to hold it up, either!

### CODE REVIEW

**Comment by kingdonb** [comment]
Verified all the links in the new file work, except for this one:

[fluxcd/community: KUBECON.md](https://github.com/fluxcd/community/blob/main/KUBECON.md)

(the file does not exist until we merge this PR)

**Comment by kingdonb** [comment]
It can be in a separate PR, or we can hold this one up and put it in the same PR, but the idea is this `KUBECON.md` will go in a filter list so changes to that one, or any of these files that are listed as [external sources](https://github.com/fluxcd/website/blob/main/external-sources/fluxcd/community) an update to those files will trigger an action in the fluxcd/website repo, so those changes trigger a build and are immediately reflected on the website after they merge:

https://how.wtf/run-workflow-step-or-job-based-on-file-changes-github-actions.html#run-job-based-on-file-changes-in-github-actions

We'll use this:

https://medium.com/hostspaceng/triggering-workflows-in-another-repository-with-github-actions-4f581f8e0ceb#:~:text=Set%20Up%20the%20Trigger%20Workflow,for%20triggering%20the%20target%20workflow.

to trigger the fluxcd/website netlify deploy job.

I think there is no third step, unless we decide that we want to add a workflow that we can update the GitHub Sites page and it filters down to this repo, then you can merge changes direct to the website just by clicking the "publish" button and you can skip the PR reviews, like you mentioned in case of some last-minute change that needs to go in immediately.

**Comment by kingdonb** [comment]
I had some time to think over how that workflow will work, and how we can expand this.

We can add additional pages and scrape them in from a source, like Google Sites, and add them as additional pages under "KubeCon!" like the "Project" header that does not have a direct link target, and is only a parent in the navbar/menu entries.

I don't know how we can make this bi-directional. If someone is editing the content in a Google Sites page, and we are scraping it into this repo, then people cannot edit the content in Git. Unless we use some kind of pull-request flow.

There is no way I can think of to get content from Git into Google Sites, unless we can POST an html / markdown content into it directly. I don't think Sites supports this kind of automation, but we can do the reverse with a GitHub Actions scheduled job and simple Ruby script. That stuff is "Phase 2" and would be in a separate, follow-on set of PRs.

**Comment by kingdonb** [comment]
TODO:

- [ ] update the kubecon.md page with content from @mewzherder Google Pages site
- [ ] automate updating the kubecon.md page similarly (with a GitHub Actions workflow)

**Comment by kingdonb** [comment]
We might need to fix some links so they do not refer back to the Google Sites page, mainly image references I mean

I think those will necessarily be PRs to fluxcd/website, so we can add the static image content there. We can combine that with fluxcd/website#1804 - I'll leave the image references intact here, we will work it out in the next PR so there is no hot-linking or cross site anything.

**Comment by kingdonb** [comment]
I found `html2md`: https://github.com/suntong/html2md#usage that we can use to script this and build the automation

```bash
wget https://sites.google.com/view/flux-kubecon-paris-2024/home -O kubecon.html \
  && html2md -i kubecon.html |sed '1,6d'|ghead -n -13 \
  > KUBECON.md \
  && rm kubecon.html
```

This is the code that we will run in the workflow, I'm building a GHA file now that will let us use `html2md` in the community repo.

There is a branch now in the community repo, `kubecon-automation`, that shows the output of the script, it isn't a perfect match but it comes close. I will iterate on that script until we have it ready to merge, but that can be in a separate PR.

It only takes 2 minutes to hand craft one of these files, and we're probably going to merge from it 5 times at most, so I don't want to spend too much time on scripting...

Happily, someone has already written html2md in go and it looks nearly suitable 🎉

**Comment by kingdonb** [comment]
I did some testing of the workflow here:

* https://github.com/kingdonb/community/pull/2

(this is an example of the pull request that will be opened when the workflow completes)

The workflow itself is here:
https://github.com/kingdonb/community/blob/main/.github/workflows/update-kubecon.yaml

I'm still working on this, it needs something still, as the go mod cache doesn't do the thing, so html2md is built every time it runs (that should be easy to fix) - I think the issue is that we don't have any `go.mod` in the repo, so maybe I can add one and that will resolve it.

I'll continue testing that workflow in the repo fork until it's ready.

**Comment by kingdonb** [comment]
I think we can merge it all at once now.

The rest of the work will be able to be done on Google Sites, and in fluxcd/website#1804

File: KUBECON.md
Comment (stefanprodan) [inline KUBECON.md:7]: What are these copy header links?

Comment (kingdonb) [reply KUBECON.md]: They are an artifact of the Google Sites page. I had manually removed them in the first iteration of the doc, that did not use html2md - they should not be in the final output so I think the script needs at least one more revision. (4ecbf0d)

They are easier to filter out now thanks to the differences in format between the previous tool and html2md


In the original Sites page, they are this handle (text shown when mouseover):
Screenshot 2024-02-22 at 3 45 26 PM

In our website version, there's no need for that, since headers are automatically turned into anchors by markdown

The script is updated so that Copy header link will be filtered out from the output.

Now that you mention it I have also added a filter now to translate the top link from `/view/flux-kubecon-paris-2024/home` to `/kubecon`

File: KUBECON.md
Comment (stefanprodan) [inline KUBECON.md:11]: The image links don't seem to work, do we need to have those in website assets maybe?

Comment (kingdonb) [reply KUBECON.md:11]: That will also make the content stable, since right now that link reference seems to change every time the Google site refreshes - it's some kind of dynamic cache key

**Comment by kingdonb** [comment]
Images are going to be a problem. As is the formatting around images. I can update the site references to use hard links so they are not 403, but we're going to have to invent some kind of system to classify them if we want them properly displayed.

**Comment by kingdonb** [comment]
This script was super annoying to test as a GitHub Action, so I broke it out into `scripts/`

I added a DEBUG flag which you can set to avoid doing the setup for html2md, in case you run this script locally and already ran go install for html2md:

```bash
export DEBUG=1
./script/update-kubecon.sh
```

The script also can't be tested on macos without additional changes because sed is not gsed

I think what we'll do is take each of the three image slug placeholders I've added with this script, and turn them into shortcodes instead. Then we can invoke whatever html we want from the hugo templates, by calling into them with alt tags.

**Comment by kingdonb** [comment]
I believe I figured out how to use shortcodes properly(ish) - I'm comfortable merging this now, I don't think the script will need more changes, unless we're adding more images.

The only thing that's going to need work now is the website repo - I've placed a CSS class on each image/figure reference, we should be able to see them in the site preview after this is merged. It's going to need a little bit of tweaking to make the CSS class do something, then we can merge fluxcd/website#1804 and make it live.

File: KUBECON.md
Comment (kingdonb) [reply KUBECON.md]: The google sites links are no longer in the output, they should be replaced by stable `figure` shortcodes which are an embedded shortcode in Hugo. 👍

**Review by kingdonb — COMMENTED** [review]
Need one approving review to merge this, @stefanprodan @mewzherder FYI

LGTM

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**
